sw/host/opentitanlib/src/transport/mod.rs

```diff
@@ -69,14 +70,26 @@ pub trait Transport {
     fn spi(&self) -> Result<Box<dyn Target>> {
         unimplemented!();
     }
+    /// Returns a mapping from symbolic names to spi port index to be used above.
+    fn enumerate_spi(&self) -> HashMap<String, u32> {
+        unimplemented!();
+    }
```

Comment:
I think you want to return a reference to the HashMap: we don't need to return a copy (and we don't expect callers to want to modify the mapping).  Also, since the mapping is probably static, you could use `HashMap<&'static str, u32>` (more on that in a later comment).